### ISSUE TITLE
How to do inference after using ImageDataGenerator for training?

### ISSUE THREAD

**Issue by brian8128** [fchollet/keras#1884]
I'll start out by saying that ImageDataGenerator is friggin' awesome for free data augmentation.  After I train the model I'm confused how to do inference though.  If I use features like zca_whitening then I will want exactly the same transformations applied to my production data where I don't know the true labels.  Is there an easy way to do this?  (Is it done automatically?)  It would be helpful if we turned on one of these transformations in the cifar10_cnn example and then expanded it to do inferencen- currently it only does training.  

https://github.com/fchollet/keras/blob/master/examples/cifar10_cnn.py

**Comment by fchollet** [comment]
You can use `ImageDataGenerator` to generate batches of data with the same characteristics as your training data, then use `model.predict_on_batch(x)` to generate predictions for these batches.

**Comment by zkself** [comment]
how to achieve this purpose in the latest keras?
